xonsh/environ.py

```diff
@@ -891,6 +912,7 @@ def current_branch(pad=True):
         branch = get_hg_branch()
     if isinstance(branch, subprocess.TimeoutExpired):
         branch = '<branch-timeout>'
+        _first_branch_timeout_message()
```

Comment:
I tried that initially, I  believe that the answer is no.  This is because the colors don't actually appear as anything, so it is easy to trigger the warning message because the color times out, without triggering the branch name timeout. It is very strange to see the warning message when you are just in your home directory and the branch name doesn't show up at all, because you are not in a repo.